### PR TITLE
Fix cronjobs and api versions

### PR DESCRIPTION
**Pull request by gchazot**
This is extracted and fixed/improved from #147 .
It's eventually a massive re-write of the logic to determine which API Version to use for CronJobs:
* Delaying the decision until the k8s cluster is contacted, unless the user specified an explicit version to use
* Using the information provided by the k8s APIs on a per-resource-kind level (the "API Groups" API doesn't give us the information we need here)

Eventually, we should drop support specifically for this case in `kubetools`, but this demonstrates how we could do it for any of the other k8s APIs in order to support multiple versions. It also helps understand the gaps in the code architecture that need to be filled in order to support this for other Resource kinds.

I've also fixed an important bug along the way: We shouldn't wait for cronjobs to finish executing on `deploy`; the next schedule might be a year from now. So we only need to wait for the CronJob resource to be created/updated

Sorry I don't find an easy way to clean-up the commit history better, so it still shows the numerous refactoring steps that were needed to clarify/simplify the code, and attempts that didn't work, until the solution became possible.

### CODE REVIEW

File: kubetools/kubernetes/api.py [hunk 1]
```diff
@@ -466,3 +309,100 @@ def check_job():
...
+    possible_versions = [
+        'batch/v1',
+        'batch/v1beta1',
+    ]
+    if default_version:
+        possible_versions.insert(0, default_version)
+
+    for version in possible_versions:
+        batch_api = _get_k8s_cronjobs_batch_api(env, version)
+        api_resources = batch_api.get_api_resources()
```
Comment (tmwoodruff) [inline kubetools/kubernetes/api.py:375]: Assume you've tested this, but does this raise an error when the cluster does not support a version. For example, when using an older k8s version that doesn't support `batch/v1`, does `get_api_resources()` return an empty list or raise an exception?

**Review by tmwoodruff — APPROVED** [review]
One small question, but otherwise looks good. Thanks for doing this!

File: kubetools/kubernetes/api.py (hunk 1 above)
Comment (gchazot) [reply kubetools/kubernetes/api.py:375]: I've tested this with k8s versions down to 1.18 and encountered no issue

**Comment by gchazot** [comment]
Thanks for the review @tmwoodruff

File: kubetools/kubernetes/api.py (hunk 1 above)
Comment (gchazot) [reply kubetools/kubernetes/api.py:375]: 100644
Added that in the release PR

**Pull request merged**